bcda/api/requests.go

```diff
@@ -514,17 +515,55 @@ func (h *Handler) bulkRequest(w http.ResponseWriter, r *http.Request, reqType se
 	}
 }
 
-func (h *Handler) validateRequest(resourceTypes []string) error {
+func (h *Handler) getResourceTypes(parameters middleware.RequestParameters, cmsID string) []string {
+	resourceTypes := parameters.ResourceTypes
+
+	// If caller does not supply resource types, we default to all supported resource types for the specific ACO
+	if len(resourceTypes) == 0 {
+		if acoConfig, found := h.Svc.GetACOConfigForID(cmsID); found {
+			if utils.ContainsString(acoConfig.Data, constants.Adjudicated) {
+				resourceTypes = append(resourceTypes, "Patient", "ExplanationOfBenefit", "Coverage")
+			}
+
+			if utils.ContainsString(acoConfig.Data, constants.PreAdjudicated) {
+				resourceTypes = append(resourceTypes, "Claim", "ClaimResponse")
+			}
+		}
+	}
+
+	return resourceTypes
+}
+
+func (h *Handler) validateRequest(resourceTypes []string, cmsID string) error {
 
 	for _, resourceType := range resourceTypes {
-		if _, ok := h.supportedResources[resourceType]; !ok {
-			return fmt.Errorf("invalid resource type %s. Supported types %s", resourceType, h.supportedResources)
+		resource, ok := h.supportedResources[resourceType]
+
+		if !ok {
+			resources := make([]string, 0, len(h.supportedResources))
+			for k := range h.supportedResources {
```

Comment:
Could we move this out (maybe to the `init()`) and avoid having to run this extra loop for every resource param? Having the slice of resource strings available might be handy elsewhere, too.